### PR TITLE
fixes widget dropdown reloading/flickering

### PR DESCRIPTION
**Pull request by astewes**
* Removes ui-sref-opts attribute, which was causing reload of the widget dropdown and preventing it from staying in focus
* Adds a FIXME note for addressing input of random strings (Metric) when creating a widget

dd

### CODE REVIEW

File: cdap-ui/app/features/dashboard/services/query-helper.js [hunk 1]
```diff
@@ -9,6 +9,10 @@ angular.module(PKG.name + '.feature.dashboard')
         // For an empty context, we want no tags. Splitting it by '.' yields [""]
         parts = [];
       }
+      // FIXME: If user enters a random context and hits Add (during widget creation)
```
Comment (anwar6953) [inline cdap-ui/app/features/dashboard/services/query-helper.js]: relevant:
https://github.com/caskdata/cdap/pull/3104

**Comment by ajainarayanan** [comment]
:+1: LGTM

**Comment by anwar6953** [comment]
We can remove the FIXME from this PR, if the following PR gets merged:
https://github.com/caskdata/cdap/pull/3104

**Comment by anwar6953** [comment]
LGTM

**Pull request merged**